File: Model/Venda.php
<?php

// Criação de uma classe, seguida de uma Herança do arquivo de conexão do banco de dados.
class Venda extends Conexao{
    
    //cria variaveis privadas para alocação de classe do venda.
    private $email;
    private $produto;
    private $idCliente;
    private $idProduto;
    private $preco;
    private $quantidade;
    private $desconto;
    private $subtotal;
    private $total;
    private $obs;
    private $data;
    private $dbQuantidade;
    private $dbSubtotal;

    //Herança da classe método buscando os camos e informaçõs da tabela no banco de dados.
    public function insert($query){
        $this->conecta();
        $this->query = $this->mysqli->query($query);
        $this->disconecta();
    }
    // Métodos Get e Set para pega e inserção de dados feitos nesta classe.
      public function setPreco($preco) {
        $this->preco= $preco;
      }

      public function getQuantidade() {
        return $this->quantidade;
      }
    
      public function setQuantidade($quantidade) {
        $this->quantidade= $quantidade;
      }

      public function getDesconto() {
        return $this->desconto;
      }
      
      public function setDesconto($desconto) {
        $this->desconto= $desconto;
      }
      public function getSubtotal() {
        return $this->subtotal;
      }
      
      //calcula o subtotal do produto com a quantidade.
      public function setSubtotal() {
        $this->subtotal = $this->quantidade * $this->dbSubtotal;
      }
      public function getTotal() {
        return $this->total;
      }
      
      //faz o calculo do total com a porcentagem de desconto aplicado.
      public function setTotal() {
        $this->total = $this->subtotal - ($this->subtotal * ($this->desconto/100)); 
      }
      public function getObs() {
        return $this->obs;
      }
      
      public function setObs($obs) {
        $this->obs= $obs;
      }
      public function getData() {
        return $this->data;
      }
      
      public function setData($data) {
        $this->data= $data;
      }
      //faz uma estrutura de condifional de se para falar ao programa que se a quantidade for maior que a quantidade ele retorna o valor dela por ela mesma para evitar bugs.
      public function getdbQuantidade() {
        if ($this->dbQuantidade >= $this->quantidade){
            return $this->dbQuantidade - $this->quantidade;
        }else{
            return $this->dbQuantidade;
        }
      }
      
      public function setdbQuantidade($dbQuantidade) {
        $this->dbQuantidade= $dbQuantidade;
      }
      public function getdbSubtotal() {
        return $this->dbSubtotal;
      }
      
      public function setdbSubtotal($dbSubtotal) {
        $this->dbSubtotal= $dbSubtotal;
      }
    
}

File: Controller/Venda.java
package br.com.liomagazine;

public class Venda {
    
    private String email;
    private String produto;
    private int idCliente;
    private int idProduto;
    private float preco;
    private int quantidade;
    private float desconto;
    private float subtotal;
    private float total;
    private String obs;
    private String data;
    private int dbQuantidade;
    private float dbSubtotal;

    public Venda(String email, String produto, float desconto, int quantidade, String obs, String data) {
        this.email = email;
        this.produto = produto;
        this.desconto = desconto;
        this.quantidade = quantidade;
        this.obs = obs;
        this.data = data;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getProduto() {
        return produto;
    }

    public void setProduto(String produto) {
        this.produto = produto;
    }

    public int getIdCliente() {
        return idCliente;
    }

    public void setIdCliente(int idCliente) {
        this.idCliente = idCliente;
    }

    public int getIdProduto() {
        return idProduto;
    }

    public void setIdProduto(int idProduto) {
        this.idProduto = idProduto;
    }

    public float getDesconto() {
        return desconto;
    }
    
    public void setDesconto(float desconto) {
        this.desconto = desconto;
    }
    
    public float getPreco() {
        return preco;
    }

    public void setPreco(float preco) {
        this.preco = preco;
    }

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }

    public float getSubtotal() {
        return subtotal;
    }

    public void setSubtotal() {
        this.subtotal = quantidade * dbSubtotal;
    }

    public float getTotal() {
        return total;
    }

    public void setTotal() {
        this.total = subtotal - (subtotal * (desconto/100));  
    }

    public String getObs() {
        return obs;
    }

    public void setObs(String obs) {
        this.obs = obs;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public int getDbQuantidade() {
        if (this.dbQuantidade >= this.quantidade){
            return this.dbQuantidade - this.quantidade;
        }else{
            return this.dbQuantidade;
        }
    }

    public void setDbQuantidade(int dbQuantidade) {
        this.dbQuantidade = dbQuantidade;
    }

    public float getDbSubtotal() {
        return dbSubtotal;
    }

    public void setDbSubtotal(float dbSubtotal) {
        this.dbSubtotal = dbSubtotal;
    }
    
}